src/orion/algo/mofa/mofa.py

```diff
@@ -0,0 +1,386 @@
+"""
+:mod:`orion.algo.mofa.mofa` -- MOFA
+============================================
+
+MOdular FActorial Design (MOFA)
+
+"""
+# pylint: disable=attribute-defined-outside-init,invalid-name
+import copy
+import logging
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+
+from orion.algo.base import BaseAlgorithm
+from orion.algo.mofa import sampler
+from orion.algo.mofa.transformer import Transformer
+from orion.algo.space import Real, Space
+from orion.core.utils.format_trials import dict_to_trial
+
+logger = logging.getLogger(__name__)
+
+
+class MOFA(BaseAlgorithm):
+    """
+    MOdular FActorial Design (MOFA).
+
+    For more information on the algorithm, see original paper: MOFA: Modular
+    Factorial Design for Hyperparameter Optimization https://arxiv.org/abs/2011.09545
+
```

Comment:
```suggestion
    Factorial Design for Hyperparameter Optimization https://arxiv.org/abs/2011.09545
    
    Xiong, Bo, Yimin Huang, Hanrong Ye, Steffen Staab, and Zhenguo Li.
    "MOFA: Modular Factorial Design for Hyperparameter Optimization."
    arXiv preprint arXiv:2011.09545 (2020).

```

Reply:
Addressed in https://github.com/Epistimio/orion/pull/864/commits/8a3e1741227b516bb43852da278a8d8725deae6d